### PR TITLE
651 Support for NFT with royalty fees 

### LINKED ISSUE #651: Support for NFT with Royalty EVM -> Hedera
image

### PR DESCRIPTION
**Pull request by vectorman1**
**Detailed description**:

**Which issue(s) this PR fixes**:
Fixes #651

**Special notes for your reviewer**:

**Checklist**
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

**Comment by failfmi** [comment]
Pull from `main` to get the latest diagrams for the flows.
Docs in https://github.com/LimeChain/hashport-validator/blob/main/docs/overview.md
https://github.com/LimeChain/hashport-validator/blob/main/docs/integration.md
have to be updated as well.

File: app/services/read-only/service.go [hunk 1]
```diff
@@ -119,6 +124,66 @@ func (s Service) FindAssetTransfer(
 	}
 }
 
+func (s Service) FindScheduledNftAllowanceApprove(
+	t *payload.Transfer,
+	sender hedera.AccountID,
+	save func(transactionID, scheduleID, status string) error) {
+	transferID := t.TransactionId
+	for {
+		txs, err := s.mirrorNode.GetTransactionsAfterTimestamp(sender, t.Timestamp.UnixNano(), ScheduleCreate)
```
Comment (failfmi) [inline app/services/read-only/service.go:133]: I think that here you will need to fetch `CryptoApproveAllowance` transactions.

File: e2e/e2e_test.go [hunk 2]
```diff
@@ -959,8 +961,8 @@ func Test_Hedera_EVM_BurnERC721_Transfer(t *testing.T) {
 		t.Fatalf("Expecting Token [%s] is not supported. - Error: [%s]", nftToken, err)
 	}
 
-	// 1. Validate that NFT owner is the bridge account
-	validateNftOwner(setupEnv, nftToken, serialNumber, setupEnv.BridgeAccount, t)
+	// 1. Validate that NFT spender is the bridge account
+	validateNftSpender(t, setupEnv, nftToken, serialNumber, setupEnv.BridgeAccount)
```
Comment (failfmi) [inline e2e/e2e_test.go:965]: I think that here the owner is still the the bridge account.

File: e2e/e2e_test.go [hunk 3]
```diff
@@ -1289,35 +1319,30 @@ func validateScheduledBurnTx(setupEnv *setup.Setup, account hedera.AccountID, as
...
 }
 
-func validateScheduledNftTransfer(setupEnv *setup.Setup, expectedTransactionID, token string, serialNum int64, t *testing.T) (transactionID, scheduleID string) {
+func validateScheduledNftAllowanceApprove(t *testing.T, setupEnv *setup.Setup, expectedTransactionID string, startTimestamp int64) (transactionID, scheduleID string) {
 	receiver := setupEnv.Clients.Hedera.GetOperatorAccountID()
 	timeLeft := 180
 
 	for {
-		response, err := setupEnv.Clients.MirrorNode.GetNftTransactions(token, serialNum)
+		scheduleCreates, err := setupEnv.Clients.MirrorNode.GetTransactionsAfterTimestamp(setupEnv.BridgeAccount, startTimestamp, read_only.ScheduleCreate)
```
Comment (failfmi) [inline e2e/e2e_test.go:1327]: read_only.CryptoApprovaAllowance

File: e2e/e2e_test.go [hunk 4]
```diff
@@ -976,13 +978,13 @@ func Test_Hedera_EVM_BurnERC721_Transfer(t *testing.T) {
...
 	// 4. Validate that a scheduled NFT transaction was submitted
-	scheduledTxID, scheduleID := validateScheduledNftTransfer(setupEnv, expectedTxId, nftToken, serialNumber, t)
+	scheduledTxID, scheduleID := validateScheduledNftAllowanceApprove(t, setupEnv, expectedTxId, blockTimestamp.UnixNano())
 
 	// 5. Validate Event Transaction ID retrieved from Validator API
 	validateEventTransactionIDFromValidatorAPI(setupEnv, expectedTxId, scheduledTxID, t)
 
 	// 6. Validate that the NFT was sent to the receiver account
-	validateNftOwner(setupEnv, nftToken, serialNumber, setupEnv.Clients.Hedera.GetOperatorAccountID(), t)
+	validateNftSpender(t, setupEnv, nftToken, serialNumber, setupEnv.Clients.Hedera.GetOperatorAccountID())
```
Comment (failfmi) [inline e2e/e2e_test.go:988]: Last but not least, the user (e2e test) has to execute the crypto transfer transaction and take the nft to its account. The idea is e2e tests to be circular - previous test wraps the NFT, this one unlocks/unwraps it.

**Review by failfmi — CHANGES_REQUESTED** [review]
.

File: docs/integration.md [hunk 5]
```diff
@@ -776,4 +776,8 @@ where `burn_erc721_id` is constructed in the following format: `{transactionHash
 | Argument            | Description                                                           |
 |---------------------|-----------------------------------------------------------------------|
 | **transactionHash** | The transaction hash of the `burnERC721` transaction                  |
-| **eventLogIndex**   | The log index of the `BurnERC721` event from the transaction receipt. |
\ No newline at end of file
+| **eventLogIndex**   | The log index of the `BurnERC721` event from the transaction receipt. |
+
+### Step 3. Transfer the native Hedera NFT back to the user
+
+After the burn transaction has been mined, the NFT is approved for the user. Then, the user can submit a transfer transaction on Hedera to get it back.
```
Comment (failfmi) [inline docs/integration.md:783]: An example of what Hedera transaction to execute using the SDK is missing.

**Review by failfmi — CHANGES_REQUESTED** [review]

**Comment by failfmi** [comment]
Tests are failing

**Review by Psykepro — APPROVED** [review]

**Review by svetlio8 — APPROVED** [review]

**Pull request merged**